sig/datadog/core/deprecations.rbs

```diff
@@ -0,0 +1,22 @@
+module Datadog
+  module Core
+    module Deprecations
```

Comment:
>> each class/module that extends Deprecations will get its own tracker of what was already warned about. I think this would be useful to include in the documentation for Deprecations.

> I don't really expect anyone to extend it.

It might be possible to enforce this via a [self type constraint](https://github.com/ruby/rbs/blob/master/docs/syntax.md#module-declaration):

> Module declaration takes optional *self type* parameter, which defines a constraint about a class when the module is mixed.

Something like (not quite sure):

```suggestion
    module Deprecations : Core
```

Reply:
Adding this with the change above causes:

```
sig/datadog/core.rbs:5:2: [error] Module self type constraint in type `::Datadog` doesn't satisfy: `singleton(::Datadog) <: ::Datadog::Core`
│ Diagnostic ID: RBS::ModuleSelfTypeError
│
└   extend Core::Deprecations
    ~~~~~~~~~~~~~~~~~~~~~~~~~
```